### PR TITLE
[ISSUE #4111] Made an explicit  null check on the map

### LINKED ISSUE #4111: [Enhancement] Do some code optimization.[AsyncHTTPPushRequest]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
image
located at:
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java line 331,343
analysis and explanation:
a. Method invocation 'get' may produce 'NullPointerException'
b. Catch Exception instead of Throwable.

### Describe the solution you'd like

a. use Objects.requireNonNull() to avoid NPE.
b. use Exception instead of Throwable.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by himansh295**
Fixes #4111 .

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java [hunk 1]
```diff
@@ -328,7 +328,7 @@ ClientRetCode processResponseContent(String content) {
             Map<String, Object> ret =
                 JsonUtils.parseTypeReferenceObject(content, new TypeReference<Map<String, Object>>() {
                 });
-            Integer retCode = (Integer) ret.get("retCode");
+            Integer retCode = (Integer) Objects.requireNonNull((ret.get("retCode")));
```
Comment (Alonexc) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java:331]: I think it should be done like this:
Integer retCode = (Integer) Objects.requireNonNull(ret).get("retCode");

Comment (himansh295) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java]: Hi @Alonexc, I tried this first but it was not giving an error. Therefore I thought to go with this way.
I have made the review changes. Can you please review it.

Comment (Alonexc) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java]: There is a suggestion to change lines 221-241 to use switch. e.g:
```
switch (result) {
                        case OK:
                        case REMOTE_OK:
                        case FAIL:
                            complete();
                            if (isComplete()) {
                                handleMsgContext.finish();
                            }
                            break;
                        case RETRY:
                        case NOLISTEN:
                            delayRetry();
                            if (isComplete()) {
                                handleMsgContext.finish();
                            }
                            break;
                        default: // do nothing
                    }
```

Comment (himansh295) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java]: Hi @Alonexc , made the required changes Please check

**Review by Alonexc — APPROVED** [review]
LGTM

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java [hunk 2]
```diff
@@ -328,7 +325,7 @@ ClientRetCode processResponseContent(String content) {
             Map<String, Object> ret =
                 JsonUtils.parseTypeReferenceObject(content, new TypeReference<Map<String, Object>>() {
                 });
-            Integer retCode = (Integer) ret.get("retCode");
+            Integer retCode = (Integer) Objects.requireNonNull(ret).get("retCode");
```
Comment (mxsm) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java:328]: How about replace "retCode" with ProtocolKey.RETCODE

Comment (himansh295) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/push/AsyncHTTPPushRequest.java]: @mxsm I have updated the required changes. Please review.

**Review by mxsm — APPROVED** [review]
LGTM

**Review by xwm1992 — APPROVED** [review]

**Review by lrhkobe — APPROVED** [review]
LGTM

**Pull request merged**